### PR TITLE
TRACING-3472: add documentation about instrumentation injection for OTEL

### PR DESCRIPTION
**Pull request by iblancasa**
Version(s):
All OCP supported versions.

Issue:
https://issues.redhat.com/browse/TRACING-3472
https://issues.redhat.com/browse/OBSDOCS-505

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: _topic_maps/_topic_map.yml [hunk 1]
```diff
@@ -3659,6 +3659,8 @@ Topics:
       File: distr-tracing-otel-installing
     - Name: Configuration
       File: distr-tracing-otel-configuring
+    - Name: Instrumentation
```
Comment (pavolloffay) [inline _topic_maps/_topic_map.yml:3662]: If we are adding this, then we should probably rename the above item to `Collector` or `Collector configuration`

File: modules/distr-tracing-otel-config-instrumentation.adoc [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+metadata:
+  name: java
+spec:
+  env:
+    - name: OTEL_TRACES_EXPORTER
+      value: otlp
+    - name: OTEL_EXPORTER_OTLP_TIMEOUT
+      value: "20"
+  exporter:
+    endpoint: http://production-collector.svc.cluster.local:4317
```
Comment (pavolloffay) [inline modules/distr-tracing-otel-config-instrumentation.adoc]: I would add namespace to the URL (e.g. `observability`) https://docs.openshift.com/container-platform/4.13/distr_tracing/distr_tracing_otel/distr-tracing-otel-using.html#distr-tracing-otel-send-traces-and-metrics-to-otel-collector-with-sidecar_distr-tracing-otel-temp

File: modules/distr-tracing-otel-config-instrumentation.adoc [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+----
+
+Table 1. Parameters used by the Operator to define the Instrumentation
+
+[cols=",,",options="header",]
+|===
+|Parameter |Description |Values
+|env |Common environment variables to define across all the instrumentations |
+|exporter |Exporter defines exporter configuration |
+|propagators |Propagators defines inter-process context propagation configuration |w3c;tracecontext;baggage;b3;none
```
Comment (pavolloffay) [inline modules/distr-tracing-otel-config-instrumentation.adoc]: Please document which one should be used for Istio

File: modules/distr-tracing-otel-config-instrumentation.adoc [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+supplementalGroups:
+  type: RunAsAny
+----
+
+[source,java]
+----
+oc adm policy add-scc-to-user otel-go-instrumentation-scc -z <SERVICE_ACCOUNT>
+----
+
+=== Configure the Python auto-instrumentation
```
Comment (pavolloffay) [inline modules/distr-tracing-otel-config-instrumentation.adoc]: Python needs to define some workarounds in the CR https://github.com/open-telemetry/opentelemetry-operator#opentelemetry-auto-instrumentation-injection

File: modules/distr-tracing-otel-config-instrumentation.adoc [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+|===
+
+To enable the injection, you need to provide the following annotation to your PodSpec:
+
+[source,java]
+----
+instrumentation.opentelemetry.io/inject-apache-httpd: "true"
+----
+
+=== Configure the DotNet auto-instrumentation
```
Comment (pavolloffay) [inline modules/distr-tracing-otel-config-instrumentation.adoc]: https://github.com/open-telemetry/opentelemetry-operator#opentelemetry-auto-instrumentation-injection 

see the comment for python

File: modules/distr-tracing-otel-config-instrumentation.adoc [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+[source,java]
+----
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: java
+spec:
+  env:
+    - name: OTEL_TRACES_EXPORTER
```
Comment (pavolloffay) [inline modules/distr-tracing-otel-config-instrumentation.adoc]: Please remove this, it is a default set by the operator

File: modules/distr-tracing-otel-config-instrumentation.adoc [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+== Instrumentation options
+
+_sample OpenTelemetry collector custom resource file_
+
+[source,java]
+----
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: java
```
Comment (pavolloffay) [inline modules/distr-tracing-otel-config-instrumentation.adoc]: a different name could be more suitable

Comment (iblancasa) [reply modules/distr-tracing-otel-config-instrumentation.adoc]: What do you think about `java-instrumentation`?

**Comment by max-cx** [comment]
I've recreated this PR as https://github.com/max-cx/openshift-docs/pull/19, where I will review it.

**Pull request closed without merging**